docs/content/docs/dev/datastream/fault-tolerance/serialization/types_serialization.md

```diff
@@ -167,6 +168,24 @@ input.keyBy(_.word)
 
 Flink supports all Java and Scala primitive types such as `Integer`, `String`, and `Double`.
 
+#### Common Collection Types
+
+Since Flink 2.0, Flink comes with dedicated serialization support for common Java collection types, which is more efficient than going
```

Comment:
It would be good to clarify here whether there are restrictions on the types allowed in collections - it is only  primitives, can it nest etc